src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/ConfigureGeneratedCodeAnalysisAnalyzer.cs

```diff
@@ -0,0 +1,130 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(CodeAnalysisDiagnosticsResources.ConfigureGeneratedCodeAnalysisDescription), CodeAnalysisDiagnosticsResources.ResourceManager, typeof(CodeAnalysisDiagnosticsResources));
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticIds.ConfigureGeneratedCodeAnalysisRuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisCorrectness,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            base.Initialize(context);
+        }
+
+        [SuppressMessage("AnalyzerPerformance", "RS1012:Start action has no registered actions.", Justification = "Method returns an analyzer that is registered by the caller.")]
+        protected override DiagnosticAnalyzerSymbolAnalyzer GetDiagnosticAnalyzerSymbolAnalyzer(CompilationStartAnalysisContext compilationContext, INamedTypeSymbol diagnosticAnalyzer, INamedTypeSymbol diagnosticAnalyzerAttribute)
+        {
+            var compilation = compilationContext.Compilation;
+
+            var analysisContext = compilation.GetTypeByMetadataName(AnalysisContextFullName);
+            if (analysisContext is null)
+            {
+                return null;
+            }
+
+            compilationContext.RegisterOperationBlockStartAction(context =>
+            {
+                if (context.OwningSymbol?.Kind != SymbolKind.Method)
+                {
+                    return;
+                }
+
+                var method = (IMethodSymbol)context.OwningSymbol;
+                IParameterSymbol analysisContextParameter = null;
+                foreach (var parameter in method.Parameters)
+                {
+                    if (!Equals(parameter.Type, analysisContext))
+                    {
+                        continue;
+                    }
+
+                    analysisContextParameter = parameter;
+                    break;
+                }
+
+                if (analysisContextParameter is null)
+                {
+                    return;
+                }
+
+                var analyzer = new ConfigureGeneratedCodeAnalyzer(analysisContextParameter);
```

Comment:
Shouldn't we also check if the method name is `Initialize`?

Reply:
Fixed